packages/kit/src/views/Staking/components/ApproveBaseStake/index.tsx

```
@@ -512,9 +500,46 @@ export function ApproveBaseStake({
+                  setApproving(false);
+                  return;
+                }
+              }
+
+              attempts += 1;
+
+              if (attempts < maxAttempts) {
+                setTimeout(checkAllowance, pollInterval);
+              } else {
+                setApproving(false);
+                return;
```

ESLint

[error] Unnecessary return statement.

(no-useless-return)